spec/factories/ext_management_system.rb

```diff
@@ -40,7 +40,13 @@
   factory :ems_physical_infra,
           :aliases => ["manageiq/providers/physical_infra_manager"],
           :class   => "ManageIQ::Providers::PhysicalInfraManager",
-          :parent  => :ext_management_system
+          :parent  => :ext_management_system do
+    trait :with_invalid_authentication do
+      after(:create) do |x|
+        x.authentications = [FactoryGirl.build(:authentication, :resource => x, :status => "invalid")]
+      end
+    end
```

Comment:
Right idea, but why inside the `:ems_physical_infra` factory? These authentications are the same for all providers, no? If you place it next to the other traits in `:ext_management_system`, this trait can then be used with all the other child factories in this file (including `:ems_physical_infra`).

Reply:
Hmm, nice!